app/assets/javascripts/backbone/views/calendar_step_modal_view.coffee

```diff
@@ -1,22 +1,32 @@
 class MS.Views.CalendarStepModalView extends Backbone.View
 
-  el: '.calendar'
+  el: '#calendar-step-modal'
+
+  initialize: (params) ->
+    @context = params.context
+    if params.id
+      @id = params.id
+      @showStep()
+    else
+      @loanId = params.loanId
+      @showNewStep()
 
   events:
-    'click .loan-calendar .cal-step': 'showStepModal'
-    'click a.action-delete': 'deleteStep'
+    'click a.action-delete': 'hideModal'
+    'click #new_project_step .cancel': 'hideModal'
+
+  showStep: ->
+    $.get '/admin/project_steps/' + @id, context: @context, (html) =>
+      @replaceContent(html)
 
-  showStepModal: (e) ->
-    calStep = e.currentTarget
-    id = @$(calStep).attr('data-step-id')
-    MS.loadingIndicator.show()
-    $.get '/admin/project_steps/' + id, context: 'calendar', (html) =>
+  showNewStep: ->
+    $.get '/admin/project_steps/new', context: @context, loan_id: @loanId, (html) =>
       @replaceContent(html)
 
   replaceContent: (html) ->
-    @$('#calendar-step-modal').find('.modal-content').html(html)
-    @$('#calendar-step-modal').modal({show: true})
+    @$el.find('.modal-content').html(html)
+    @$el.modal({show: true})
```

Comment:
I believe this initializes the modal _and then_ shows it. Do you really want to initialize it each time this method is called? Should the initializer (just `.modal()`) go in `initialize` above, and just call `modal('show')` here?